panel/src/components/Views/LanguageView.vue

```diff
@@ -87,7 +87,7 @@ export default {
 		},
 		option(option, row) {
 			this.$panel.dialog.open(
-				`dialogs/languages/${this.id}/translations/${row.key}/${option}`
+				`dialogs/languages/${this.id}/translations/${encodeURIComponent(window.btoa(row.key))}/${option}`
```

Comment:
Just `encodeURIComponent` was not enough. To support HTML tags in addition to normal strings, I developed a solution by first encoding the text with base64 and then decoding it on the PHP side.

Reply:
I'd be fine with base64, but I'm curious in which cases plain URL encoding would break. I have tried `<script>alert("XSS")</alert>` (encoded as `%3Cscript%3Ealert(%22XSS%22)%3C%2Falert%3E`) and that seemed to work.

Do you have an example for a key that would break without base64 encoding?

Reply:
![image (19)](https://github.com/getkirby/kirby/assets/3393422/07637040-089b-47b8-9b1a-d2fc2331d5e1)

`<script>alert('XSS');</script>` doesn't for me on windows local.

> Not Found
> The requested URL was not found on this server.
> Apache/2.4.46 (Win64) OpenSSL/1.1.1h PHP/8.1.10 Server at localhost Port 80

Reply:
That's weird. There must be something wrong in our router then. On macOS with Caddy and PHP 8.1 it works fine...

Could you add TODO comments to the lines where the Base64 encoding and decoding happens with a short explanation that this is necessary only for compatibility?

Reply:
Here the changes https://github.com/getkirby/kirby/pull/5370/commits/1bf1d68fb47933739ccf005bb8df2b0345ebe632

But my explanations may not be very clear. Sorry for that 🙈